controllers/provider-aws/pkg/webhook/controlplanebackup/ensurer.go

```diff
@@ -96,50 +97,56 @@ func (e *ensurer) getBackupRestoreContainer(name string, cluster *extensionscont
 	// They are only specified for the etcd-main stateful set (backup is enabled)
 	var (
 		provider                string
+		prefix                  string
 		env                     []corev1.EnvVar
 		volumeClaimTemplateName = name
 	)
 	if name == gardencorev1alpha1.StatefulSetNameETCDMain {
-		provider = aws.StorageProviderName
-		env = []corev1.EnvVar{
-			{
-				Name: "STORAGE_CONTAINER",
-				// The bucket name is written to the backup secret by Gardener as a temporary solution.
-				// TODO In the future, the bucket name should come from a BackupBucket resource (see https://github.com/gardener/gardener/blob/master/docs/proposals/02-backupinfra.md)
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						Key:                  aws.BucketName,
-						LocalObjectReference: corev1.LocalObjectReference{Name: aws.BackupSecretName},
+		if cluster.Seed.Spec.Backup == nil {
+			e.logger.Info("Backup profile is not configured; backups will not be taken for etcd-main")
+		} else {
+			prefix = common.GenerateBackupEntryName(cluster.Shoot.Status.TechnicalID, cluster.Shoot.Status.UID)
+			provider = aws.StorageProviderName
+			env = []corev1.EnvVar{
+				{
+					Name: "STORAGE_CONTAINER",
+					// The bucket name is written to the backup secret by Gardener as a temporary solution.
+					// TODO In the future, the bucket name should come from a BackupBucket resource (see https://github.com/gardener/gardener/blob/master/docs/proposals/02-backupinfra.md)
```

Comment:
Can this (and all similar) TODOs be removed?

Reply:
Ok. Sure. Once gardener/gardener#1128 is merged. No dependency on it. But already that PR is huge. And once we have something in it would be easy to progress.

Reply:
Alright, sounds good, thank you!